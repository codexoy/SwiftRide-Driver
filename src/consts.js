import { strings } from "./i18n"

export const TOKEN_KEY = "login_token"
export const REFRESH_TOKEN_KEY = "refresh_token"
export const NOTIFICATIONS_KEY = "notifications"
export const ROOT_URL = "https://dryvo.herokuapp.com"
export const API_DATE_FORMAT = "YYYY-MM-DDTHH:mm:ss.SSS"
export const SHORT_API_DATE_FORMAT = "YYYY-MM-DD"
export const DISPLAY_SHORT_DATE_FORMAT = "DD.MM.YYYY"
export const DATE_FORMAT = "DD/MM/YYYY"
export const DEFAULT_IMAGE =
	"https://res.cloudinary.com/hsmkon3br/image/upload/c_thumb,g_face,h_80,w_80/v1554188012/678099-profile-filled-256.png"
export const DEFAULT_IMAGE_MAX_SIZE = 200
export const DEFAULT_MESSAGE_TIME = 1000
export const STORAGE_PREFIX = "Dryvo_"
export const DEFAULT_ERROR = strings("default_error")
export const MAIN_PADDING = 26
export const DEFAULT_DURATION = 40
export const NAME_LENGTH = 16
export const colors = {
	blue: "rgb(12,116,244)",
	green: "rgb(24, 199, 20)"
}
export const calendarTheme = {
	textSectionTitleColor: "#000",
	dayTextColor: "#000",
	textDisabledColor: "rgb(155,155,155)",
	textDayFontFamily: "Assistant",
	textMonthFontFamily: "Assistant",
	textDayHeaderFontFamily: "Assistant",
	todayTextColor: "#000",
	selectedDayTextColor: "#fff",
	selectedDayBackgroundColor: colors.blue,
	textMonthFontWeight: "bold",
	textDayFontWeight: "500",
	selectedDayfontWeight: "bold",
	textDayFontSize: 16,
	textMonthFontSize: 16,
	textDayHeaderFontSize: 16
}
export const floatButtonOnlyStyle = {
	backgroundColor: colors.blue,
	width: 160,
	height: 56,
	borderRadius: 28,
	alignItems: "center",
	justifyContent: "center",
	shadowColor: colors.blue,
	shadowOffset: {
		width: 0,
		height: 8
	},
	shadowOpacity: 0.5,
	shadowRadius: 16,
	elevation: 8
}
export const fullButton = {
	...floatButtonOnlyStyle,
	position: "absolute",
	bottom: 0,
	width: "100%",
	borderRadius: 0
}
