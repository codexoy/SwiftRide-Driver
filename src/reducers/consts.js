export const LOAD_FETCH_SERVICE = "LOAD_FETCH_SERVICE"
export const LOGIN = "LOGIN"
export const LOGOUT = "LOGOUT"
