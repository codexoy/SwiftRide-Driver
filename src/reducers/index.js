import { combineReducers } from "redux"
import userReducer from "./user"
import fetchServiceReducer from "./fetch"
import errorsReducer from "./errors"

const appReducer = combineReducers({
	user: userReducer,
	fetchService: fetchServiceReducer,
	errors: errorsReducers
})
export default appReducer
