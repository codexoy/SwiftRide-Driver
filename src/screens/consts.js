import React from "react"
import { Icon } from "react-native-elements"
const options = {
	activeTintColor: "rgb(12,116,244)",
	inactiveTintColor: "black",
	showLabel: false
}

export const tabBarOptions = {
	defaultNavigationOptions: ({ navigation }) => ({
		tabBarIcon: ({ focused, horizontal, tintColor }) => {
			const { routeName } = navigation.state
			let iconName
			let iconType = "material"
			if (routeName === "Home") {
				iconName = "home"
			} else if (routeName === "Notifications") {
				iconName = "notifications"
			} else if (routeName === "Lesson") {
				iconName = "add-circle"
			} else if (routeName === "Students") {
				iconName = "people"
			} else if (routeName == "Schedule") {
				iconName = "date-range"
			} else if (routeName == "Profile") {
				iconName = "person"
			}

			return (
				<Icon
					name={iconName}
					size={25}
					color={tintColor}
					type={iconType}
				/>
			)
		}
	}),
	tabBarOptions: options
}
