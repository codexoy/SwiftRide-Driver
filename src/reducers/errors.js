import { API_ERROR, POP_ERROR } from "./consts"
import { APIError } from "../error_handling"

apiClassName = APIError.name

initialErrors = {
	[apiClassName]: [],
	other: []
} // more errors to come

export default function errorsReducer(state = initialErrors, action) {
	switch (action.type) {
		case API_ERROR:
			return Object.assign({}, state, {
				[apiClassName]: [...state[apiClassName], action.error]
			})
		case POP_ERROR:
			// pop one of the arrays (action.errorType)
			const afterLastPopped = state[action.errorType].filter(
				(v, i) => i != state[action.errorType].length - 1
			) // take all but last
			return Object.assign({}, state, {
				[action.errorType]: afterLastPopped
			})
		default:
			return state
	}
}
