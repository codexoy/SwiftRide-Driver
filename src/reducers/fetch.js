import * as consts from "./consts"
import Fetch from "../services/Fetch"

export default function fetchServiceReducer(state = null, action) {
	switch (action.type) {
		case consts.LOAD_FETCH_SERVICE:
			return new Fetch()
		default:
			return state
	}
}
