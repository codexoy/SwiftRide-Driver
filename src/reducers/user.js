import { LOGIN, LOGOUT, CHANGE_USER_IMAGE } from "./consts"

export default function userReducer(state = null, action) {
	switch (action.type) {
		case LOGIN:
			return action.user
		case LOGOUT:
			return null
		case CHANGE_USER_IMAGE:
			return { ...state, image: action.image }
		default:
			return state
	}
}
