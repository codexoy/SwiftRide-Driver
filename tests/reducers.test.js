import user from "../src/reducers/user"
import errors from "../src/reducers/errors"
import fetch from "../src/reducers/fetch"
import * as consts from "../src/reducers/consts"

describe("user reducer", () => {
	it("should return the initial state", () => {
		expect(user(undefined, {})).toEqual(null)
	})

	it("should handle LOGIN - return user object", () => {
		expect(
			user(null, {
				type: consts.LOGIN,
				user: {
					name: "test"
				}
			})
		).toMatchSnapshot()
	})

	it("should handle LOGOUT - return null", () => {
		expect(
			user(null, {
				type: consts.LOGOUT
			})
		).toMatchSnapshot()
	})

	it("should handle CHANGE_USER_IMAGE - return user object with `test2` image", () => {
		expect(
			user(
				{ image: "test1" },
				{ type: consts.CHANGE_USER_IMAGE, image: "test2" }
			)
		).toMatchSnapshot()
	})
})

describe("errors reducer", () => {
	it("should return the initial state - return dict of empty arrays with the correct keys", () => {
		expect(errors(undefined, {})).toMatchSnapshot()
	})
	const initialState = {
		[consts.API_ERROR]: [],
		other: []
	}
	it("should handle API ERROR - insert `test error` to array", () => {
		expect(
			errors(initialState, {
				type: consts.API_ERROR,
				error: "test error"
			})
		).toMatchSnapshot()
	})

	it("should handle API ERROR - insert default message if empty is given", () => {
		expect(
			errors(initialState, {
				type: consts.API_ERROR,
				error: ""
			})
		).toMatchSnapshot()
	})

	it("should handle APP ERROR - insert `test inside app error` to array", () => {
		expect(
			errors(undefined, {
				type: consts.APP_ERROR,
				error: "test inside app error"
			})
		).toMatchSnapshot()
	})

	const filledWithErrors = Object.assign({}, initialState, {
		[consts.API_ERROR]: ["test 1", "test 2"]
	})
	it("should handle POP ERROR and pop from array - return only `test 1` in array", () => {
		expect(
			errors(filledWithErrors, {
				type: consts.POP_ERROR,
				errorType: consts.API_ERROR
			})
		).toMatchSnapshot()
	})
})

describe("fetch reducer", () => {
	it("should return the initial state", () => {
		expect(fetch(undefined, {})).toEqual(null)
	})

	it("should handle LOAD FETCH SERVICE - return `Fetch` object", () => {
		expect(
			fetch(null, {
				type: consts.LOAD_FETCH_SERVICE
			})
		).toMatchSnapshot()
	})
})
