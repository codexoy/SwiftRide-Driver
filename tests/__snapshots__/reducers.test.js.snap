// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors reducer should handle API ERROR - insert \`test error\` to array 1`] = `
Object {
  "API_ERROR": Array [
    "test error",
  ],
  "other": Array [],
}
`;

exports[`errors reducer should handle API ERROR - insert default message if empty is given 1`] = `
Object {
  "API_ERROR": Array [
    "קרתה תקלה, אנא נסו שנית.",
  ],
  "other": Array [],
}
`;

exports[`errors reducer should handle APP ERROR - insert \`test inside app error\` to array 1`] = `
Object {
  "API_ERROR": Array [],
  "APP_ERROR": Array [
    "test inside app error",
  ],
  "other": Array [],
}
`;

exports[`errors reducer should handle POP ERROR and pop from array - return only \`test 1\` in array 1`] = `
Object {
  "API_ERROR": Array [
    "test 1",
  ],
  "other": Array [],
}
`;

exports[`errors reducer should return the initial state - return dict of empty arrays with the correct keys 1`] = `
Object {
  "API_ERROR": Array [],
  "APP_ERROR": Array [],
  "other": Array [],
}
`;

exports[`fetch reducer should handle LOAD FETCH SERVICE - return \`Fetch\` object 1`] = `
Fetch {
  "_RESEND": Symbol(RESEND),
  "_definedExceptions": Object {
    "BLACKLISTED_TOKEN": "_handleExpiredToken",
    "EXPIRED_TOKEN": "_handleExpiredToken",
  },
  "_requestsLimit": 2,
  "defaultHeaders": Object {
    "Accept": "application/json",
    "Content-Type": "application/json",
  },
  "sentRequests": 0,
}
`;

exports[`user reducer should handle CHANGE_USER_IMAGE - return user object with \`test2\` image 1`] = `
Object {
  "image": "test2",
}
`;

exports[`user reducer should handle LOGIN - return user object 1`] = `
Object {
  "name": "test",
}
`;

exports[`user reducer should handle LOGOUT - return null 1`] = `null`;
